api/client.go

```diff
@@ -689,7 +689,9 @@ START:
 	}
 
 	if timeout != 0 {
-		ctx, _ = context.WithTimeout(ctx, timeout)
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, timeout)
```

Comment:
Thanks for pointing that out.  You're right: back in #4987 you removed an existing call to the cancelfunc that was already there.  Some background is given in #4973 and #4979, where the idea of doing it in a defer was discussed.  It looks like that approach was rejected because of @jefferai 's comment that 

> The problem with deferring is that we support retries so these could pile up

but that confuses me because there's at best a single retry in this scope (namely, when redirectCount==0).